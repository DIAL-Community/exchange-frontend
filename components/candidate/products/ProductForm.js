import React, { useContext, useState } from 'react'
import { useIntl } from 'react-intl'
import { useRouter } from 'next/router'
import { useMutation } from '@apollo/client'
import { FaSpinner } from 'react-icons/fa'
import ReCAPTCHA from 'react-google-recaptcha'
import { Controller, useForm } from 'react-hook-form'
import { useSession } from 'next-auth/client'
import { ToastContext } from '../../../lib/ToastContext'
import { CREATE_CANDIDATE_PRODUCT } from '../../../mutations/product'
import Input from '../../shared/Input'
import ValidationError from '../../shared/ValidationError'
import { HtmlEditor } from '../../shared/HtmlEditor'
import { emailRegex } from '../../shared/emailRegex'
import { Unauthorized } from '../../shared/FetchStatus'

const ProductForm = () => {
  const { formatMessage } = useIntl()
  const format = (id, values) => formatMessage({ id }, { ...values })

  const router = useRouter()

  const [session] = useSession()

  const { showToast } = useContext(ToastContext)

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const [createCandidateProduct] = useMutation(CREATE_CANDIDATE_PRODUCT, {
    onError: () => {
      showToast(
        format('candidate-product.submit.failure'),
        'error',
        'top-center',
        false
      )
    },
    onCompleted: () => {
      showToast(
        format('candidate-product.submit.success'),
        'success',
        'top-center',
        1000,
        null,
        () => router.push('/products')
      )
    }
  })

  const { handleSubmit, register, control, formState: { errors } } = useForm({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: '',
      description: '',
      repository: '',
      website: '',
      email: '',
      captcha: null
    }
  })

  const doUpsert = async (data) => {
    if (session) {
      setMutating(true)

      const { userEmail, userToken } = session.user
      const { name, description, repository, website, email, captcha } = data
      const variables = {
        name,
        website,
        repository,
        description,
        email,
        captcha
      }

      createCandidateProduct({
        variables,
        context: {
          headers: {
            'Accept-Language': router.locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  const cancelForm = () => {
    setReverting(true)
    router.push('/products')
  }

  return session ? (
    <div className='flex flex-col'>
      <div className='py-8 px-8'>
        <div id='content' className='sm:px-0 max-w-full mx-auto'>
          <form onSubmit={handleSubmit(doUpsert)}>
            <div className='bg-edit shadow-md rounded px-8 pt-6 pb-12 mb-4 flex flex-col gap-3'>
              <div className='text-2xl font-bold text-dial-blue pb-4'>
                {format('candidateProduct.label')}
              </div>
              <div className='flex flex-col lg:flex-row gap-4'>
                <div className='w-full lg:w-1/2 flex flex-col gap-y-3'>
                  <div className='form-field-wrapper' data-testid='candidate-product-name'>
                    <label className='form-field-label required-field' htmlFor='name'>
                      {format('candidateProduct.name')}
                    </label>
                    <Input
                      {...register('name', { required: format('validation.required') })}
                      id='name'
                      placeholder={format('candidateProduct.name.placeholder')}
                      isInvalid={errors.name}
                    />
                    {errors.name && <ValidationError value={errors.name?.message} />}
                  </div>
                  <div className='form-field-wrapper' data-testid='candidate-product-website'>
                    <label className='form-field-label' htmlFor='website'>
                      {format('candidateProduct.website')}
                    </label>
                    <Input
                      {...register('website')}
                      id='website'
                      placeholder={format('candidateProduct.website.placeholder')}
                    />
                  </div>
                  <div className='form-field-wrapper' data-testid='candidate-product-repository'>
                    <label className='form-field-label' htmlFor='repository'>
                      {format('candidateProduct.repository')}
                    </label>
                    <Input
                      {...register('repository')}
                      id='repository'
                      placeholder={format('candidateProduct.repository.placeholder')}
                    />
                  </div>
                  <div className='form-field-wrapper' data-testid='candidate-product-email'>
                    <label className='form-field-label required-field' htmlFor='email'>
                      {format('candidateProduct.email')}
                    </label>
                    <Input
                      type='email'
                      {...register('email', {
                        required: format('validation.required'),
                        pattern: { value: emailRegex , message: format('validation.email') }
                      })}
                      id='email'
                      placeholder={format('candidateProduct.email.placeholder')}
                      isInvalid={errors.email}
                    />
                    {errors.email && <ValidationError value={errors.email?.message} />}
                  </div>
                  <Controller
                    name='captcha'
                    control={control}
                    rules={{ required: format('validation.required') }}
                    render={({ field: { onChange, ref } }) => {
                      return (<ReCAPTCHA sitekey={process.env.NEXT_PUBLIC_CAPTCHA_SITE_KEY} ref={ref} onChange={onChange} />)
                    }}
                  />
                  {errors.captcha && <ValidationError value={errors.captcha?.message} />}
                </div>
                <div className='w-full lg:w-2/3' style={{ minHeight: '20rem' }} data-testid='candidate-product-description'>
                  <label className='block text-xl text-dial-blue flex flex-col gap-y-2'>
                    <p className='required-field'> {format('candidateProduct.description')}</p>
                    <Controller
                      name='description'
                      control={control}
                      rules={{ required: format('validation.required') }}
                      render={({ field: { value, onChange, onBlur } }) => {
                        return (
                          <HtmlEditor
                            editorId={`${name}-editor`}
                            onBlur={onBlur}
                            onChange={onChange}
                            initialContent={value}
                            isInvalid={errors.description}
                            placeholder={format('candidateProduct.description.placeholder')}
                          />
                        )
                      }}
                    />
                    {errors.description && <ValidationError value={errors.description?.message} />}
                  </label>
                </div>
              </div>
              <div className='flex flex-wrap font-semibold text-xl lg:mt-8 gap-3'>
                <button
                  type='submit'
                  data-testid='submit-button'
                  className='submit-button'
                  disabled={mutating || reverting}
                >
                  {format('candidateProduct.submit')}
                  {mutating && <FaSpinner className='spinner ml-3 inline' />}
                </button>
                <button
                  type='button'
                  className='cancel-button'
                  disabled={mutating || reverting}
                  onClick={cancelForm}
                >
                  {format('app.cancel')}
                  {reverting && <FaSpinner className='spinner ml-3 inline' />}
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  ) : <Unauthorized />
}

export default ProductForm
