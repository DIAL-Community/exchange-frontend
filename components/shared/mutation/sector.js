import { gql } from '@apollo/client'

export const DELETE_SECTOR = gql`
  mutation DeleteSector($id: ID!) {
    deleteSector(id: $id) {
      sector {
       id
       slug
       name
      }
      errors
    }
  }
`

export const CREATE_SECTOR = gql`
  mutation CreateSector(
    $name: String!
    $slug: String!
    $isDisplayable: Boolean!
    $locale: String!
  ) {
    createSector(
      name: $name
      slug: $slug
      isDisplayable: $isDisplayable
      locale: $locale
    ) {
      sector {
        id
        name
        slug
        locale
        isDisplayable
      }
      errors
    }
  }
`
