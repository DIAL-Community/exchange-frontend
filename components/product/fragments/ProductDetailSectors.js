import { useCallback, useContext, useState } from 'react'
import { useRouter } from 'next/router'
import { useIntl } from 'react-intl'
import { useApolloClient, useMutation } from '@apollo/client'
import { ToastContext } from '../../../lib/ToastContext'
import EditableSection from '../../shared/EditableSection'
import Pill from '../../shared/form/Pill'
import Select from '../../shared/form/Select'
import { UPDATE_PRODUCT_SECTORS } from '../../shared/mutation/product'
import { SECTOR_SEARCH_QUERY } from '../../shared/query/sector'
import { fetchSelectOptions } from '../../utils/search'

const ProductDetailSectors = ({ product, editingAllowed }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const client = useApolloClient()

  const [sectors, setSectors] = useState(product.sectors)
  const [isDirty, setIsDirty] = useState(false)

  const { locale } = useRouter()

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const [updateProductSectors, { loading, reset }] = useMutation(UPDATE_PRODUCT_SECTORS, {
    onError() {
      setIsDirty(false)
      setSectors(product?.sectors)
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.sector.header') }))
      reset()
    },
    onCompleted: (data) => {
      const { updateProductSectors: response } = data
      if (response?.product && response?.errors?.length === 0) {
        setIsDirty(false)
        setSectors(response?.product?.sectors)
        showSuccessMessage(format('toast.submit.success', { entity: format('ui.sector.header') }))
      } else {
        setIsDirty(false)
        setSectors(product?.sectors)
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.sector.header') }))
        reset()
      }
    }
  })

  const fetchedSectorsCallback = (data) => (
    data.sectors?.map((sector) => ({
      id: sector.id,
      name: sector.name,
      slug: sector.slug,
      label: sector.name
    }))
  )

  const addSector = (sector) => {
    setSectors([
      ...[
        ...sectors.filter(({ id }) => id !== sector.id),
        { id: sector.id, name: sector.name, slug: sector.slug  }
      ]
    ])
    setIsDirty(true)
  }

  const removeSector = (sector) => {
    setSectors([...sectors.filter(({ id }) => id !== sector.id)])
    setIsDirty(true)
  }

  const onSubmit = () => {
    updateProductSectors({
      variables: {
        sectorSlugs: sectors.map(({ slug }) => slug),
        slug: product.slug
      },
      context: {
        headers: {
          'Accept-Language': locale
        }
      }
    })
  }

  const onCancel = () => {
    setSectors(product.sectors)
    setIsDirty(false)
  }

  const displayModeBody = sectors.length
    ? <div className='flex flex-col gap-y-2'>
      {sectors?.map((sector, index) =>
        <div key={`sector-${index}`}>
          <div key={index}>{sector.name}</div>
        </div>
      )}
    </div>
    : <div className='text-sm text-dial-stratos'>
      {format('general.na')}
    </div>

  const sectionHeader =
    <div className='font-semibold text-dial-meadow'>
      {format('ui.sector.header')}
    </div>

  const editModeBody =
    <div className='px-4 lg:px-6 py-4 flex flex-col gap-y-3 text-sm'>
      <label className='flex flex-col gap-y-2'>
        {`${format('app.searchAndAssign')} ${format('ui.sector.label')}`}
        <Select
          async
          isSearch
          isBorderless
          defaultOptions
          cacheOptions
          placeholder={format('shared.select.autocomplete.defaultPlaceholder')}
          loadOptions={(input) =>
            fetchSelectOptions(client, input, SECTOR_SEARCH_QUERY, fetchedSectorsCallback)
          }
          noOptionsMessage={() => format('filter.searchFor', { entity: format('ui.sector.label') })}
          onChange={addSector}
          value={null}
        />
      </label>
      <div className='flex flex-wrap gap-3'>
        {sectors.map((sector, sectorIdx) => (
          <Pill
            key={`sectors-${sectorIdx}`}
            label={sector.name}
            onRemove={() => removeSector(sector)}
          />
        ))}
      </div>
    </div>

  return (
    <EditableSection
      editingAllowed={editingAllowed}
      sectionHeader={sectionHeader}
      onSubmit={onSubmit}
      onCancel={onCancel}
      isDirty={isDirty}
      isMutating={loading}
      displayModeBody={displayModeBody}
      editModeBody={editModeBody}
    />
  )
}

export default ProductDetailSectors
