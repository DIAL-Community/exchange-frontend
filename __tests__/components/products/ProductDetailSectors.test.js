import { fireEvent, screen, waitFor } from '@testing-library/react'
import ProductDetailSectors from '../../../components/products/ProductDetailSectors'
import { SECTOR_SEARCH_QUERY } from '../../../queries/sector'
import { mockRouterImplementation, mockSessionImplementation, render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { sectors } from './data/ProductDetailSectors'
import { product } from './data/ProductForm'

// Mock next-router calls.
jest.mock('next/dist/client/router')
// Mock the next-auth's useSession.
jest.mock('next-auth/client')

describe('Unit test for the ProductDetailSectors component.', () => {
  const EDIT_BUTTON_TEST_ID = 'edit-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const SECTOR_SEARCH_TEST_ID = 'sector-search'
  const SECTOR_SEARCH_PLACEHOLDER = 'Type to search...'
  const SECTOR_SEARCH_OPTION_1_LABEL = 'Sector 1'
  const SECTOR_SEARCH_OPTION_2_LABEL = 'Sector 2'
  const PRODUCT_TEST_SECTOR_LABEL = 'Test Sector'
  const PILL_TEST_ID = 'pill'
  const PILL_REMOVE_BUTTON_TEST_ID = 'remove-button'
  const mockSectors = generateMockApolloData(SECTOR_SEARCH_QUERY, { search: '', locale: 'en' }, null, sectors)

  beforeAll(() => {
    mockRouterImplementation()
    mockSessionImplementation()
  })

  test('Should match snapshot - without edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockSectors]}>
        <ProductDetailSectors
          canEdit={false}
          product={product}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockSectors]}>
        <ProductDetailSectors
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with open editable section', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockSectors]}>
        <ProductDetailSectors
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )

    await waitFor(() => new Promise((res) => setTimeout(res, 0)))

    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(SECTOR_SEARCH_PLACEHOLDER)
    await waitFor(() => expect(container.querySelector('.react-select__loading-indicator')).toBeNull())
    expect(container).toMatchSnapshot()
  })

  test('Should remove a pill', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockSectors]}>
        <ProductDetailSectors
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(SECTOR_SEARCH_PLACEHOLDER)
    fireEvent.click(getByTestId(PILL_REMOVE_BUTTON_TEST_ID))
    expect(screen.queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should add a pill and revert changes on "Cancel" button click', async () => {
    const { container, getByTestId, getByText } = render(
      <CustomMockedProvider mocks={[mockSectors]}>
        <ProductDetailSectors
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(SECTOR_SEARCH_PLACEHOLDER)

    fireEvent.keyDown(getByTestId(SECTOR_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(SECTOR_SEARCH_OPTION_1_LABEL)
    fireEvent.click(getByText(SECTOR_SEARCH_OPTION_1_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(2)

    fireEvent.keyDown(getByTestId(SECTOR_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(SECTOR_SEARCH_OPTION_2_LABEL)
    fireEvent.click(getByText(SECTOR_SEARCH_OPTION_2_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(3)

    expect(container).toMatchSnapshot()

    fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
    expect(screen.queryByText(PRODUCT_TEST_SECTOR_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(SECTOR_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(SECTOR_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()

    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(SECTOR_SEARCH_PLACEHOLDER)

    expect(screen.queryByText(PRODUCT_TEST_SECTOR_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(SECTOR_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(SECTOR_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(1)
  })
})
