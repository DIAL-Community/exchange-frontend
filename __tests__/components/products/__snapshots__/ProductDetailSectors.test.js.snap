// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the ProductDetailSectors component. Should add a pill and revert changes on "Cancel" button click 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Sectors
      </div>
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Sectors
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="sector-search"
      >
        Search and Assign Sectors
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-4-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-4-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Sector
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Sector 1
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Sector 2
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the ProductDetailSectors component. Should match snapshot - with edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Sectors
      </div>
      <button
        class="mr-2 inline-flex items-center gap-x-1.5 bg-dial-blue px-2 py-1 rounded-md text-white"
        data-testid="edit-button"
        type="button"
      >
        <img
          alt="Edit"
          height="12px"
          src="/icons/edit.svg"
          width="12px"
        />
        <span
          class="text-sm"
        >
          Edit
        </span>
      </button>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-2"
    >
      <div
        class="border-3 border-transparent"
        data-testid="sector-card"
      >
        <div
          class="flex justify-between border border-dial-gray card-drop-shadow px-4 h-16"
        >
          <div
            class="flex items-center p-4"
          >
            <div
              class="inline-block font-semibold text-button-gray"
            >
              Test Sector
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the ProductDetailSectors component. Should match snapshot - with open editable section 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Sectors
      </div>
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Sectors
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="sector-search"
      >
        Search and Assign Sectors
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-2-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Sector
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the ProductDetailSectors component. Should match snapshot - without edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Sectors
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-2"
    >
      <div
        class="border-3 border-transparent"
        data-testid="sector-card"
      >
        <div
          class="flex justify-between border border-dial-gray card-drop-shadow px-4 h-16"
        >
          <div
            class="flex items-center p-4"
          >
            <div
              class="inline-block font-semibold text-button-gray"
            >
              Test Sector
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the ProductDetailSectors component. Should remove a pill 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Sectors
      </div>
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Sectors
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="sector-search"
      >
        Search and Assign Sectors
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-19jej77-control"
          >
            <div
              class="react-select__value-container css-3tzzl5-ValueContainer"
            >
              <div
                class="react-select__placeholder css-19jynah-placeholder"
                id="react-select-3-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      />
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
